EcosystemSampleApp/SampleAppViewController.swift

```diff
@@ -339,5 +342,61 @@ class SampleAppViewController: UIViewController, UITextFieldDelegate {
             }
         }
     }
+    
+    @IBAction func orderConfirmationTapped(_ sender: Any) {
+        guard let offerId = lastOfferId else{
+            
+            let alert = UIAlertController(title: "No Order has Been Made", message: "No Order was sent in this session yet.", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "Oh ok", style: .cancel, handler: { [weak alert] action in
+                alert?.dismiss(animated: true, completion: nil)
+            }))
+            self.present(alert, animated: true, completion: nil)
+            
+            return
+        }
+        print("orderConfirmationTapped offerId : \(offerId)")
+        guard let appId = appId, let jwtPKey = privateKey else {
+            alertConfigIssue()
+            return
+        }
+        
+        do {
+            try jwtLoginWith(lastUser, id: appId)
+        } catch {
+            alertStartError(error)
+        }
+        
+        Kin.shared.orderConfirmation(for: offerId) { (status, err) in
+            DispatchQueue.main.async { [weak self] in
```

Comment:
@yosriz a good practise is to create a `strongSelf` guard when only needing to execute code while the view controller is active.